### PR TITLE
test: Improvements to wait_for_pods function

### PR DESCRIPTION
**Pull request by hbelmiro**
This PR replaces the bash function that checks the readiness of the KFP deploy with a Python script that relies on the Kubernetes API.

Resolves: https://github.com/kubeflow/pipelines/issues/11160
Resolves: https://github.com/kubeflow/pipelines/issues/11161

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: scripts/deploy/github/kfp-readiness/wait_for_pods.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            else:
+                stable_count = 0
+        else:
+            stable_count = 0
+
+        previous_statuses = current_statuses
+        logging.info(f"Pods are still stabilizing. Retrying in {calm_time} seconds...")
+        time.sleep(calm_time)
+    else:
+        logging.info("Pods did not stabilize within the timeout period.")
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply scripts/deploy/github/kfp-readiness/wait_for_pods.py]: Done.

File: scripts/deploy/github/kfp-readiness/wait_for_pods.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        sys.exit(1)
+
+    logging.info("Final pod statuses:")
+    for pod_name, (pod_status, ready, total) in previous_statuses.items():
+        if pod_status == 'Running' and ready == total:
+            logging.info(f"Pod {pod_name} is fully ready ({ready}/{total})")
+        else:
+            logging.info(f"Pod {pod_name} is not ready (Status: {pod_status}, Ready: {ready}/{total})")
+
+    sys.exit(0)
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply scripts/deploy/github/kfp-readiness/wait_for_pods.py]: Done.

**Comment by hbelmiro** [comment]
@diegolovison @DharmitD can you please review?

**Review by DharmitD — APPROVED** [review]
/lgtm
/Approve

**Pull request merged**